2023 Update

Asingh11771 opened the pull request:
Update to new WPILIB

Asingh11771 committed:
Tank Drive

Basic Tank Drive

Asingh11771 committed:
Fast Mode and SlewRateLimiters

Fast Mode Command and SlewRateLimiters

Asingh11771 committed:
Fast Mode and Slow Mode

Fast Mode and Slow Mode Commands, Fast is set to START and Slow is set to BACK both with toggle functionality

Asingh11771 committed:
Arcade Drive

Asingh11771 committed:
Drive Voltage and Current

Asingh11771 committed:
Camera

Asingh11771 committed:
PhotonLib Import

PhotonLib Import

Asingh11771 committed:
Drive Speeds

Tank and Arcade Drive Speeds

Asingh11771 committed:
Vision Align

Vision Align using PhotonVision

Asingh11771 committed:
Camera Subsystem

Camera Subsystem

Asingh11771 committed:
Camera SmartDashboard

Camera SmartDashboard

Asingh11771 committed:
Vision Constants

Asingh11771 committed:
Sims and 2023 Update

alwick commented on src/main/java/frc/robot/commands/toggleFastMode.java:
```diff
@@ -0,0 +1,39 @@
+
+
+public class toggleFastMode extends CommandBase {
+  /** Creates a new toggleFastModer. */
```
Classes should always start with Cap.

alwick commented on src/main/java/frc/robot/commands/toggleSlowMode.java:
```diff
@@ -0,0 +1,35 @@
+
+
+public class toggleSlowMode extends CommandBase {
+  /** Creates a new toggleFastModer. */
```
Classes start with cap.

Asingh11771 committed:
Function Capitalization

Function Capitalization

Asingh11771 committed:
Swerve Base

Swerve Base

Asingh11771 committed:
Clean Up

Clean Up

Asingh11771 committed:
Clean Up 2

Clean Up 2

Asingh11771 committed:
AHRS Gyro

AHRS Gyro

alwick committed:
limelight

limelight

BluAutumn committed:
gyro fix

alwick committed:
limelight Streaming

limelight Streaming

Asingh11771 committed:
NavX Gyro Lib

NavX Gyro Lib

arovge commented on README.md:
```diff
@@ -1,2 +0,0 @@
-# RobotCode2023
```
We can keep this if you want - could maybe include some info on the robot and such towards the end of the season

arovge commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -206,4 +207,10 @@ public static class IntakeConstants {
     public static final int kIntake = 20;
 
   }
+  public static class VisionConstants {
```
Could move this to a separate file - maybe make a `constants` folder and make a `constants/VisionConstants.java` file in it. Would help quickly finding the constant you want to tweak instead of having a file with a few hundred lines you need to scan through

arovge commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -56,19 +66,112 @@
+
   // The robot's subsystems
+
+  private final DriveSubsystem m_robotDrive = new DriveSubsystem();
```
I made this comment when I was porting the wiki over to github: https://github.com/Team537/frc-wiki/pull/1#discussion_r1055962519

I don't mind if y'all want to keep using `m_*` or `k*` when writing code since a lot of the WPILIB code has it built in, but I think it takes away from readability

arovge commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -56,19 +66,112 @@
+// The driver controllers
+
   XboxController m_driverController = new XboxController(OIConstants.kDriverControllerPort);
   XboxController m_driverController2 = new XboxController(OIConstants.kDriverControllerPort);
```
Be consistent with access modifiers - some of these have `private` and some don't have any modifier

arovge commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -56,19 +66,112 @@
+    //Toggle Booleans
+
+    boolean toggleFastMode = true;
+    boolean toggleSlowMode = true;
```
I think these vars and the forwardSpeed / turnSpeed should be in DriveSubsystem - it's implementation details of the drive subsystem

arovge commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -56,19 +66,112 @@
+   //Drive Commands
+
+    m_robotDrive.setDefaultCommand(
+      // new RunCommand(() ->
```
remove the commented out code - i guess it doesn't matter because this will change to be swerve drive but

arovge commented on src/main/java/frc/robot/commands/toggleFastMode.java:
```diff
@@ -0,0 +1,39 @@
+  public ToggleFastMode(DriveSubsystem drive) {
+    this.drive = drive;
+    addRequirements(drive);
+    // Use addRequirements() here to declare subsystem dependencies.
```
Could remove these comments if you want - up to you

arovge commented on src/main/java/frc/robot/subsystems/Camera.java:
```diff
@@ -0,0 +1,34 @@
+package frc.robot.subsystems;
```
Spacing is messed up in this file

arovge commented on src/main/java/frc/robot/subsystems/DriveSubsystem.java:
```diff
@@ -0,0 +1,437 @@
+package frc.robot.subsystems;
```
Spacing messed up in this file & some extra comments you could get rid of (up to you)

alwick committed:
Changing Pipelines

Changing Pipelines

alwick committed:
Library Update

Library Update

Asingh11771 committed:
Alliance April Tag Pipelines

Alliance April Tag Pipelines

Asingh11771 committed:
Alliance String

Alliance String

alwick committed:
Button Rebindings

Button Rebindings

alwick committed:
Swerve Slow Mode

Swerve Slow Mode

BluAutumn committed:
Remove Toggle fast/slow mode, add gripper commands + buttons

alwick committed:
Lib Update and Gripper code

Lib Update and Gripper code

BluAutumn committed:
update photon

BluAutumn committed:
remove errors with last years toggle fast/slow mode, update navx and phoenix

BluAutumn committed:
Merge branch 'Gripper' into development

Asingh11771 committed:
remove slow mode

remove slow mode

Asingh11771 committed:
Initial

Gasterblaster805 committed:
Gripper Port Change

Gripper Port Change

Asingh11771 committed:
Swerve Sim, Not working

Swerve Sim, Not working

Asingh11771 committed:
Attempted Fix

Asingh11771 committed:
Revert Swerve Base

Revert Swerve Base

Asingh11771 committed:
Revert "Revert Swerve Base"

This reverts commit 5cf878757168d9ff642f56ed8b415ab8cbe7dc7d.

Asingh11771 committed:
Working Swerve Sim

Working Swerve Sim

Asingh11771 committed:
Swerve 2.0

Swerve 2.0

Asingh11771 committed:
Drive Volts Sim Tuned

Drive Volts Sim Tuned

Asingh11771 committed:
Swerve 3.0 Working Simulation

Swerve 3.0 Working Simulation

Asingh11771 committed:
Merge branch 'Swerve-Sim' into swerve

Asingh11771 committed:
Simulation Tweaks

Simulation Tweaks

Asingh11771 committed:
Field Image

Asingh11771 committed:
Slow Mode

Slow Mode

Asingh11771 committed:
Slow Mode Tweaks

Slow Mode Tweaks

Asingh11771 committed:
Initial Pose

Initial Pose

Asingh11771 committed:
Path Planner Example Trajectory

Path Planner Example Trajectory

Asingh11771 committed:
Rot Turn Tuned

Rot Turn Tuned

Gasterblaster805 committed:
Blue Auto 1

Gasterblaster805 committed:
Blue Auto 2+3

Gasterblaster805 committed:
Blue Auto Markers 1-4

Asingh11771 committed:
Follow Trajectory with Markers

Asingh11771 committed:
Util

Asingh11771 committed:
Rot Max Speed Change

Asingh11771 committed:
Auto Chooser

Asingh11771 committed:
Swerve test

Swerve test

BluAutumn committed:
make new commands

Asingh11771 committed:
Pigeon Implementation

Pigeon Implementation

Asingh11771 committed:
Util 2

Asingh11771 committed:
gyro

gyro

Asingh11771 committed:
2023

2023

Asingh11771 committed:
SRX Mag encoder Fix

SRX Mag encoder Fix

Asingh11771 committed:
Motor Ids

Motor Ids

Asingh11771 committed:
Mag Encoder Class

Mag Encoder Class

Asingh11771 committed:
Update SwerveModule.java

encoder reset

BluAutumn committed:
update comments

Asingh11771 committed:
srx mag encoder reset code

srx mag encoder reset code

Asingh11771 committed:
ArmInOut NEO

Asingh11771 committed:
start pid

Asingh11771 committed:
working swerve?

Front Modules X, Rear O

Asingh11771 committed:
SmartDashboard

SmartDashboard

Asingh11771 committed:
Increment

Increment

Asingh11771 committed:
Refactoring

Refactoring

Asingh11771 committed:
Refactoring 2

Refactoring 2

Asingh11771 committed:
robot centric swerve

robot centric swerve

Asingh11771 committed:
Working Field Centric

Working Field Centric

Asingh11771 committed:
Merge branch 'swerve-testing' into development

Asingh11771 committed:
Merge branch 'limelight-testing' into development

Asingh11771 committed:
fix f

fix f

Asingh11771 committed:
Merge branch 'Arm-in/out-final(neo)' into development

Asingh11771 committed:
sure

sure

Asingh11771 committed:
Wrist

Asingh11771 committed:
Gripper Tested

Gripper Tested

Asingh11771 committed:
Merge branch 'Gripper' into development

Asingh11771 committed:
fixg

fixg

Asingh11771 committed:
offest

offest

Asingh11771 committed:
update

update

Asingh11771 committed:
config

config

Asingh11771 committed:
swerbe

Asingh11771 committed:
Merge branch 'Wrist' into development

Asingh11771 committed:
merge wrist into dev

Asingh11771 committed:
add wrist and pivot buttons

Asingh11771 committed:
try setting the drive to coast to allow deceleration

BluAutumn committed:
Update gradle

BluAutumn committed:
add dist per rot magencoder

Asingh11771 committed:
Swerve Accel / Decel Testing plus Sussex Scrimmage Code

Swerve Accel / Decel Testing plus Sussex Scrimmage Code

Asingh11771 committed:
Merge branch 'development' of https://github.com/Team537/RobotCode2023 into development

Asingh11771 committed:
meg encodar

meg encodar

Asingh11771 committed:
Intake Camera

Intake Camera

Asingh11771 committed:
Slew Rate Limiter

Slew Rate Limiter

Asingh11771 committed:
Revert "meg encodar"

This reverts commit 7e5ca7ceb8f12696b7c9148ad985c235ef047f27.

Asingh11771 committed:
Revert "Slew Rate Limiter"

This reverts commit 66d3830319dd52b291ab8370c9cfc7cf76e189aa.

Asingh11771 committed:
Revert "Intake Camera"

This reverts commit 7a60971efb94d761c4a4522ed90622c3a5bc24a9.

Asingh11771 committed:
Working acceleration, non tuned

Asingh11771 committed:
Gripper Camera

Gripper Camera

Asingh11771 committed:
Pose Estimator

Pose Estimator

Asingh11771 committed:
AprilTag Maybe?

AprilTag Maybe?

BluAutumn committed:
LED

BluAutumn committed:
Merge branch 'development' of https://github.com/Team537/RobotCode2023 into development

Asingh11771 committed:
started tuning accel decel

BluAutumn committed:
LED Yellow And Purple

Asingh11771 committed:
accel/decel

accel/decel

Asingh11771 committed:
Merge branch 'development' of https://github.com/Team537/RobotCode2023 into development

Asingh11771 committed:
add lights,

BluAutumn committed:
led changes

Asingh11771 committed:
swerveb

swerveb

Asingh11771 committed:
Merge branch 'development' of https://github.com/Team537/RobotCode2023 into development

BluAutumn committed:
update

BluAutumn committed:
Merge branch 'development' of https://github.com/Team537/RobotCode2023 into development

BluAutumn committed:
Added Manipulator buttons, Changed arm inout and pivot set pos constants, removed cobra intake

BluAutumn committed:
edit formatiing

Asingh11771 committed:
Mag Encoder Working

Mag Encoder Working

Asingh11771 committed:
Merge branch 'mag-encoder' of https://github.com/Team537/RobotCode2023 into mag-encoder

Asingh11771 committed:
fix manip

fix manip

Asingh11771 committed:
Merge branch 'mag-encoder' into development

BluAutumn committed:
Variables for set pos

Asingh11771 committed:
Pose Estimator (Unused As of Now)

Asingh11771 committed:
Very Cool LEDS

Asingh11771 committed:
More Cool LED stuff

Asingh11771 committed:
LED and Manipulator Revamp

Asingh11771 committed:
Commented out LEDS for testing Manipulator

Asingh11771 committed:
s

Asingh11771 committed:
Drive LEDS

Asingh11771 committed:
Commented Out LEDS

Asingh11771 committed:
Drive Slow actually Drives Slow

Asingh11771 committed:
Abs Value

Asingh11771 committed:
Organized Folders

Asingh11771 committed:
WIP Manipulator Commands 2.0

Asingh11771 committed:
Manipulator commands wip

Asingh11771 committed:
Manipulator Commands 2.0

Asingh11771 committed:
Commented Out

Asingh11771 committed:
limelight

limelight

Asingh11771 committed:
intake cam

intake cam

BluAutumn committed:
Update test

BluAutumn committed:
Merge branch 'development' of https://github.com/Team537/RobotCode2023 into development

BluAutumn committed:
add test ground and zero for each manipulator

BluAutumn committed:
change buttons to new commands

Asingh11771 committed:
shelf fix

Asingh11771 committed:
intake cam change

Asingh11771 committed:
Merge branch 'PathPlanner2' into development

Asingh11771 committed:
Auto Merge

Asingh11771 committed:
Gripper Camera New File

Asingh11771 committed:
pipeline

Asingh11771 committed:
blinkin update

lol

Asingh11771 committed:
Set Position fixes

BluAutumn committed:
test values, started reorganizeing commands

BluAutumn committed:
Merge branch 'development' of https://github.com/Team537/RobotCode2023 into development

Asingh11771 committed:
Merge remote-tracking branch 'origin/development' into development

Asingh11771 committed:
Fixed Slow Mode

BluAutumn committed:
Add new shelf commands

BluAutumn committed:
Merge branch 'development' of https://github.com/Team537/RobotCode2023 into development

BluAutumn committed:
started adding zero pos

Asingh11771 committed:
Field Constants

Asingh11771 committed:
Merge remote-tracking branch 'origin/development' into development

Asingh11771 committed:
merge fixes

BluAutumn committed:
Zero commands and buttons

BluAutumn committed:
Merge branch 'development' of https://github.com/Team537/RobotCode2023 into development

BluAutumn committed:
format commands

BluAutumn committed:
remove unused imports

BluAutumn committed:
Remove shelf high and change shelf mid to human player

Asingh11771 committed:
blinkin without button presets

Asingh11771 committed:
blinkin semi-full update

Asingh11771 committed:
blinkin full update version 1.0

BluAutumn committed:
night work

Asingh11771 committed:
LED blinkin update 1.1

Asingh11771 committed:
blinkin led update 1.2

Asingh11771 committed:
blinkin LED update 2.0

- NOW STOPS COMMANDS FROM FIGHTING IN EPIC WARS.

- CLEANED BLINKIN CODE, AND SIMPLIFIED IT.

BluAutumn committed:
update today

BluAutumn committed:
change commands to start end

BluAutumn committed:
Wrist StartEnd

BluAutumn committed:
Update RobotContainer.java

BluAutumn committed:
Wrist Adjustment

Asingh11771 committed:
updates after practice

BluAutumn committed:
update after practice2

Asingh11771 committed:
Merge branch 'Objective-Tracker' into development

BluAutumn committed:
after practice 3

BluAutumn committed:
Merge branch 'development' of https://github.com/Team537/RobotCode2023 into development

Asingh11771 committed:
ScoreMidDriveBack Auto

Asingh11771 committed:
Boost Mode (needs to be tuned)

Asingh11771 committed:
LED Changes

Asingh11771 committed:
auto changes

Asingh11771 committed:
boost mode changes

Asingh11771 committed:
Auto Open Loop Change

Asingh11771 committed:
boost mode changes

BluAutumn committed:
change wrist manual down to pivot manual down

BluAutumn committed:
Changed arminout and pivot pidf to new constants locations

BluAutumn committed:
Removed old manipulator subsystem, switch wrist to new constants, and delete pidf test commands

BluAutumn committed:
delete bolt+cobra drive motor constants and commented the constants

Asingh11771 committed:
tuesday 7th changes

Asingh11771 committed:
Merge branch 'development' of https://github.com/Team537/RobotCode2023 into development

Asingh11771 committed:
Reverting Boost Mode Changes

BluAutumn committed:
fix names of commands

BluAutumn committed:
comment for knowing what things are
